### ISSUE TITLE
Classes with accents on description fail since switch to UTF-8

### ISSUE THREAD

**Issue by isaacsb** [latex3/latex2e#66]
## Brief outline of the bug

After making UTF-8 the default input encoding for LaTeX files, classes using accents on the description have stopped working with the "classic" compiler. The minimal example works correctly under previous versions (1.40.18 and older) or the unicode compilers (LuaLaTeX and XeLaTeX).

Tested on Linux (TeX Live 2015-2018), Windows (latest MiKTeX) and Mac (MacTeX/TL 2018).

## Minimal example showing the bug

```latex
\RequirePackage{latexbug}       % <--should be always the first line (see CONTRIBUTING)!
\documentclass{myclass}
\begin{document}
Hello
\end{document}
```

`myclass.cls`:
```latex
\NeedsTeXFormat{LaTeX2e}
\ProvidesClass{myclass}[2018/08/10 v0.1 Descripción]
\LoadClass{article}
```

## Log file (required) and possibly PDF file  
[test.log](https://github.com/latex3/latex2e/files/2279180/test.log)

**Comment by FrankMittelbach** [comment]
The  arguments of the`\Provides...` in LaTeX never officially supported anything other than 7bit ascii. The fact that most of the time \ProvidesClass would work (as in your example above) doesn't mean that it is correct or supported syntax. Two counter examples:

`myencclass.cls`:

```
 \RequirePackage[utf8]{inputenc}  % or some other input encoding
\LoadClass{myclass}
```
Here your class will die, and the above is clearly official proper LaTeX.

Or any `\ProvidesPackage` , `...File` or what have you in the preamble after you loaded inputenc.

Fact is, latex2e was designed with 7bit syntax due to the restrictions posed by 8bit engines as 7bit is the only common subset among different 8bit encodings (one of which is in some sense utf8). 

So your class example works because at that point there isn't yet any encoding support yet, but as my example shows your class could be used in a different way too. And once there is encoding support all bets are off as by design the characters are typeset which gives you nice results like
```
Package: mypackage 2018/08/10 v0.1 Descripci\unhbox \voidb@x \bgroup \let \unhb
ox \voidb@x \setbox \@tempboxa \hbox {o\global \mathchardef \accent@spacefactor
 \spacefactor }\accent 19 o\egroup \spacefactor \accent@spacefactor n
```
or fatal errors like loops or ... depending on the characters used if you try it on package level.

unicode engines are quite different  as they do not handle uft8 characters as sequences of single bytes but they see the real characters. This is a central difference and one that means that  À in luatex is something totally different than À in pdftex and there is only so much to mask that. One my the way is to restrict the code syntax to 7bit.

So in summary, works as designed and still does, ie unsupported cases may change over time, which is not necessary nice in an individual case but unavoidable really.

### Enhancements

Having said all that, we are aware about the fact that with our move to better utf8 support for typesetting there is also the need to extend the support for utf8 (or a reasonable subset) in  other places where this is possible. And one such place is or will be something like terminal typeout (which is what `\Provides...` is about. But this is not so trivial if this should really work well in the end, which is why we haven't tried to put anything into the kernel in a rush. Eventually this will come.

**Comment by isaacsb** [comment]
Good to know, it didn't occur to me that it was a side effect of not having a default encoding. I guess I will have to remove the accents from the info and warning messages in my class.

I find interesting that this doesn't work either:
```latex
\NeedsTeXFormat{LaTeX2e}
\ProvidesClass{myclass}[2018/08/10 v0.1 Descripci\'on]
\LoadClass{article}
```

Is the reason the same?

**Comment by isaacsb** [comment]
Oops, isn't this a duplicate of #52? I don't know why I didn't see it yesterday.

**Comment by FrankMittelbach** [comment]
> I find interesting that this doesn't work either:
> 
> \NeedsTeXFormat{LaTeX2e}
> \ProvidesClass{myclass}[2018/08/10 v0.1 Descripci\'on]
> \LoadClass{article}
> 
> Is the reason the same?

yes, same reason. \typeout and the like really do the same a typesetting 
text (but only as far as expansion is concerned, so assignments etc 
aren't happening) and for straight a ascii chars that is then "e" -> "e" 
but for something like ó or \'o they expand to the same thing and that 
then breaks.

In the end the solution has to be that in such cases the encoding is 
changed to result in a suitable output representation for the terminal 
(or for a bookmark or ...) then it would start to work, but as I said we 
aren't there yet.

I'll close this issue and any progress can be tracked on #52
